lib/portage/gpkg.py

```diff
@@ -347,9 +347,9 @@ def kill(self):
         kill external program if any error happened in python
         """
         if self.proc is not None:
-            self.killed = True
-            self.proc.kill()
             self.proc.stdin.close()
+            self.proc.kill()
+            self.killed = True
```

Comment:
This code exists twice in the file. It should also be changed at the other place.

Reply:
Done, thanks